### PR TITLE
Enable dts-trimming for packages/dds

### PR DESCRIPTION
**Pull request by jikim-msft**
#### Description

Related: [18176](https://github.com/microsoft/FluidFramework/pull/18176/files)

This PR adds type-trimming for `packages/dds` directory. The changes introduced are: 
- Add npm `copyfiles` package as `devDependency`
- Add `dtsRollup` in `api-extractor.json`
- Enable both type-trimming for both `ESM` & `CJS`
- Make `fluidBuild` to fetch dependencies from the scripts included in `dependsOn`

### CODE REVIEW

File: packages/dds/test-dds-utils/src/ddsFuzzHarness.ts [hunk 1]
```diff
@@ -153,7 +183,11 @@ function getSaveInfo(
  *     validateConsistency: (a, b) => { assert.equal(a.getText(), b.getText()); }
  * }
  * ```
- * This model can be used directly to create a suite of fuzz tests with {@link createDDSFuzzSuite}
+ * This model can be used directly to create a suite of fuzz tests with
```
Comment (Josmithr) [inline packages/dds/test-dds-utils/src/ddsFuzzHarness.ts:186]: Nit: I'm not sure we need the line break here

File: packages/dds/test-dds-utils/src/ddsFuzzHarness.ts [hunk 2]
```diff
@@ -153,7 +183,11 @@ function getSaveInfo(
...
+ * {@link (createDDSFuzzSuite:function)}
+ *
```
Comment (Josmithr) [inline packages/dds/test-dds-utils/src/ddsFuzzHarness.ts:188]: Nit
```suggestion
```

File: packages/dds/test-dds-utils/src/ddsFuzzHarness.ts [hunk 3]
```diff
@@ -357,7 +397,7 @@ export interface DDSFuzzSuiteOptions {
 	 * ```
 	 *
 	 * @remarks
-	 * If you prefer, a variant of the standard `.only` syntax works. See {@link createDDSFuzzSuite.only}.
+	 * If you prefer, a variant of the standard `.only` syntax works. See {@link (createDDSFuzzSuite:function)}.
```
Comment (Josmithr) [inline packages/dds/test-dds-utils/src/ddsFuzzHarness.ts:400]:
```suggestion
	 * If you prefer, a variant of the standard `.only` syntax works. See {@link (createDDSFuzzSuite:namespace).only}.
```

File: packages/dds/test-dds-utils/src/ddsFuzzHarness.ts [hunk 4]
```diff
@@ -1117,7 +1164,8 @@ export async function replayTest<
 }
 
 /**
- * Creates a suite of eventual consistency tests for a particular DDS model.
+ * TODO: Write documentation.
```
Comment (Josmithr) [inline packages/dds/test-dds-utils/src/ddsFuzzHarness.ts:1167]:
```suggestion
 * Creates a suite of eventual consistency tests for a particular DDS model.
```

File: packages/dds/test-dds-utils/src/ddsFuzzHarness.ts [hunk 5]
```diff
@@ -1088,6 +1131,10 @@ function isInternalOptions(options: DDSFuzzSuiteOptions): options is InternalOpt
 	return options.only instanceof Set && options.skip instanceof Set;
 }
 
+// eslint-disable-next-line jsdoc/require-description
```
Comment (Josmithr) [inline packages/dds/test-dds-utils/src/ddsFuzzHarness.ts:1134]: Is this rule disable needed? It doesn't look like we need it elsewhere in this package.

File: packages/dds/test-dds-utils/src/gcTestRunner.ts [hunk 6]
```diff
@@ -36,6 +38,11 @@ export interface IGCTestProvider {
 	addNestedHandles(): Promise<void>;
 }
 
+// eslint-disable-next-line jsdoc/require-description
+/**
+ *
```
Comment (Josmithr) [inline packages/dds/test-dds-utils/src/gcTestRunner.ts:43]:
```suggestion
```

File: packages/dds/test-dds-utils/src/gcTestRunner.ts [hunk 7]
```diff
@@ -36,6 +38,11 @@ export interface IGCTestProvider {
 	addNestedHandles(): Promise<void>;
 }
 
+// eslint-disable-next-line jsdoc/require-description
```
Comment (Josmithr) [inline packages/dds/test-dds-utils/src/gcTestRunner.ts:41]: Is this needed?

File: packages/dds/test-dds-utils/src/index.ts [hunk 8]
```diff
@@ -19,3 +19,4 @@ export {
 	Synchronize,
 	replayTest,
 } from "./ddsFuzzHarness";
+export { MinimizationTransform } from "./minification";
```
Comment (Josmithr) [inline packages/dds/test-dds-utils/src/index.ts:22]: Note for reviewers: this is transitively referenced by an existing export and therefor needs to be exported.

**Review by Josmithr — APPROVED** [review]
Left a few comments, but otherwise looks good!

Nit for the PR title: this PR doesn't really enable the trimming, it just enables the type-rollup generation.

File: packages/dds/test-dds-utils/src/ddsFuzzHarness.ts [hunk 9]
```diff
@@ -1197,43 +1245,51 @@ const getFullModel = <TChannelFactory extends IChannelFactory, TOperation extend
...
- * @example
- *
- * ```typescript
- * // Runs only seed 42 for the given model.
- * createDDSFuzzSuite.only(42)(model);
- * ```
+ * {@inheritDoc (createDDSFuzzSuite:function)}
+ * @internal
  */
-createDDSFuzzSuite.only =
```
Comment (Josmithr) [inline packages/dds/test-dds-utils/src/ddsFuzzHarness.ts:1209]: Note for reviewers: this syntax implicitly creates a namespace called `createDDSFuzzSuite` under the hood, which we were not exporting, which caused API-Extractor to complain.

File: packages/dds/test-dds-utils/src/ddsFuzzHarness.ts (hunk 5 above)
Comment (jikim-msft) [reply packages/dds/test-dds-utils/src/ddsFuzzHarness.ts:1132]: image

The disable is needed as the linter is complaining about the missing JSDoc. This is same for the `test-dds-utils/gcTestRunner.ts`.

File: packages/dds/test-dds-utils/src/gcTestRunner.ts (hunk 7 above)
Comment (jikim-msft) [reply packages/dds/test-dds-utils/src/gcTestRunner.ts:41]: image

Explained above. I could also leave a brief description for the functions (`replayTest` & `runGCTests`) out of my understanding (or could leave `TODO`).

File: packages/dds/test-dds-utils/src/ddsFuzzHarness.ts (hunk 5 above)
Comment (Josmithr) [reply packages/dds/test-dds-utils/src/ddsFuzzHarness.ts]: Okay, just wanted to double check. Some of our packages have this rule enabled, and some don't. Looks good!

**Pull request merged**